os/Kconfig

```diff
@@ -198,6 +198,42 @@ config BUILD_KERNEL
 
 endchoice # Build configuration
 
+config FLASH_START_ADDR
+	hex "Flash memory start address"
+	default 0x0
+	---help---
+		Start address of Flash memory
+
+if BUILD_PROTECTED
+
+config SRAM_START_ADDR
+	hex "RAM memory start address"
+	default 0x0
+	---help---
+		Start address of RAM memory to be used by the linker script. If multiple RAM are available,
+		then this config gives the start address of the RAM region for which the kernel and user
+		binaries need to be linked during the build process.
+
+config KSRAM_SIZE
+	int "Kernel RAM size in MB"
+	default 0
+	---help---
+		Size of kernel RAM region in KB.
+
+config USRAM_SIZE
+	int "User RAM size in MB"
+	default 0
+	---help---
+		Size of user RAM region in KB.
+
+config AUTOGEN_MEMORY_LDSCRIPT
```

Comment:
Could you make a depenedency between board and this?
Because imxrt1020-evk and imxrt1050-evk should have this with protected build.
```
config ARCH_BOARD_IMXRT1050_EVK
	bool "NXP i.MX RT 1050 EVK"
	depends on ARCH_CHIP_FAMILY_IMXRT105x
	select ARCH_HAVE_LEDS
	select ARCH_HAVE_BUTTONS
	select ARCH_HAVE_IRQBUTTONS
	select AUTOGEN_MEMORY_LDSCRIPT if BUILD_PROTECTED  <--- like this
```

Reply:
ok. Done